packages/shared/src/types/config/dev.ts

```diff
@@ -76,6 +82,10 @@ export interface DevConfig {
    * Used to control whether the build artifacts of the development environment are written to the disk.
    */
   writeToDisk?: boolean | ((filename: string) => boolean);
+  /**
+   * This option allows you to configure a list of globs/directories/files to watch for file changes.
+   */
+  watchFiles?: string | string[] | WatchFiles;
```

Comment:
Can you add this new config to documentation~

Reply:
<img width="329" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/25540709/80e808c3-d2a4-40f0-af6e-1de1ac0c9120">
Do you mean like this, add the introduction and use of this feature to the documentation?

Reply:
yes, just like other config options

Reply:
ok, no problem.

Reply:
Hi, i finished the documentation, you can review it.